src/wasm.h

```diff
@@ -183,42 +193,48 @@ struct Literal {
       case WasmType::i32: return i32 == other.i32;
       case WasmType::i64: return i64 == other.i64;
       // reinterpret floating-point, to avoid nan != nan
-      case WasmType::f32: return reinterpreti32() == other.reinterpreti32();
-      case WasmType::f64: return reinterpreti64() == other.reinterpreti64();
+      case WasmType::f32: {
+        if (std::isnan(f32) || std::isnan(other.f32)) return floatPayload() == other.floatPayload();
```

Comment:
Added signbit.

Yes, this code avoids comparing nans, because they always compare false.